### PR TITLE
Copy original FRED initialization code

### PR DESCRIPTION
**Pull request by asarium**
This copies the original FRED code for initializing the individual game
systems to qtFRED. This should fix the various issues caused by missing
game system initialization calls.

### CODE REVIEW

File: qtfred/src/main.cpp [hunk 1]
```diff
@@ -119,38 +113,60 @@ int main(int argc, char* argv[]) {
...
+								 { SubSystem::Parse,             app.tr("Initializing parser") },
+								 { SubSystem::TechroomIntel,     app.tr("Initializing techroom intel") },
+								 { SubSystem::Nebulas,           app.tr("Initializing nebulas") },
+								 { SubSystem::Stars,             app.tr("Initializing stars") },
+								 { SubSystem::Ssm,               app.tr("Initializing SSMs") },
+								 { SubSystem::EventMusic,        app.tr("Initializing event music") },
+								 { SubSystem::FictionViewer,     app.tr("Initializing fiction viewer") },
+								 { SubSystem::CommandBriefing,   app.tr("Initializing command briefing") },
+								 { SubSystem::Campaign,          app.tr("Initializing campaign system") },
+								 { SubSystem::NebulaLighting,    app.tr("Initializing nebula lighting") },
```
Comment (jg18) [inline qtfred/src/main.cpp]: Maybe I misread things, but shouldn't this be NebulaLightning?

File: qtfred/src/mission/management.cpp [hunk 2]
```diff
@@ -26,75 +35,220 @@ extern void allocate_mission_text(size_t size);
...
+	Xstr_inited = 1;
+
+#ifndef NDEBUG
 	load_filter_info();
 #endif
 
-	std::setlocale(LC_ALL, "C");
+	//CFREDView *window = CFREDView::GetView();
+	//HWND hwndApp = window->GetSafeHwnd();
+	//os_set_window((uint) hwndApp);
```
Comment (jg18) [inline qtfred/src/mission/management.cpp]: Can you remove this commented out code if we don't need it?

**Comment by jg18** [comment]
I tested these changes with my fiction viewer dialog branch and the fiction music options now appear correctly in the drop down box. Cool stuff. Just a couple small comments.

File: qtfred/src/main.cpp (hunk 1 above)
Comment (asarium) [reply qtfred/src/main.cpp]: It seems like I misunderstood what that system was for...

File: qtfred/src/mission/management.cpp (hunk 2 above)
Comment (asarium) [reply qtfred/src/mission/management.cpp]: Fixed.

**Pull request merged**